src/cryptography/x509.py

```diff
@@ -1158,3 +1225,18 @@ def signature_hash_algorithm(self):
         Returns a HashAlgorithm corresponding to the type of the digest signed
         in the certificate.
         """
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RevokedCertificate(object):
```

Comment:
Ok, I'll add the extensions property to the RevokedCertificate interface as well. However, at the moment I'm not planning to support any CRL extensions in this putback. They are fairly complicated and I personally have never needed them, but I can add them in a later pull request. 
In the meantime, I could just return an empty list if you want me to add extensions as a required property to the CRL class.

I might need to add the CRL extensions anyway to properly support the CRL.verify() functionality. As I just found out, it's not enough to compare if the issuers of cert and CRL match, sometimes the issuer is in the Revoked extension 'certificateIssuer'. However, you can only evaluate this extension if the CRL has an extensions which marks it as indirect.
So currently I punted on it and don't compare CRL issuer and cert issuer before I pass the two to X509_CRL_verify().